osu.Game/Beatmaps/IBeatmap.cs

```diff
@@ -69,6 +70,43 @@ public interface IBeatmap
         /// </summary>
         double GetMostCommonBeatLength();
 
+        double AudioLeadIn { get; internal set; }
```

Comment:
I guess my biggest concern is requiring these in the `IBeatmap` interface (and the delegation requirement of implementations). Which is probably not a huge concern at all.

I'm neutral on moving stuff into a separate class, will require further discussion to ascertain what we're looking for here, as touched on in other comment thread.